new-player/src/main/java/net/newpipe/newplayer/NewPlayerImpl.kt

```diff
@@ -501,7 +501,7 @@ internalPlayStream(mediaSource: MediaSource, playMode: PlayMode) {
         val builder = MediaSourceBuilder(
             repository = repository,
             mutableErrorFlow = mutableErrorFlow,
-            httpDataSourceFactory = repository.getHttpDataSourceFactory(item),
+            dataSourceFactory = repository.getDataSourceFactory(item, app),
```

Comment:
```suggestion
            dataSourceFactory = repository.getDataSourceFactory(item),
```

Reply:
A repository will have the ability to access a `Context` regardless of weather that `Context` was supplied from the interface call or not. So this should be removed eventually.